code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -1092,6 +1092,10 @@ var/global/list/damage_icon_parts = list()
 		O.name = "[index]"
 		O.icon = t_inhand_state
 		O.icon_state = t_state
+		if(I.color)
+			O.color = I.color
+		else
+			O.color = null // this prevents a bug where all inhands would be colored after holding a colored item
```

Comment:
Yes, it should be redundant, but not having it causes the bug. I havn't looked closely at the overlay code but I think it's because the overlays are cached somewhere.